eq-author-api/middleware/identification/upsertUser.test.js

```diff
@@ -33,4 +38,23 @@ describe("Upsert User", () => {
     });
     expect(res.json).toHaveBeenCalledWith({ status: "OK" });
   });
+
+  it("should stop unallowed emails from being saved", async () => {
+    process.env.ALLOWED_EMAIL_LIST = "@a.fake.com";
+    await upsertUser(req, res, next);
+    const user = await getUserByExternalId(req.user.externalId);
+    expect(user).toBeUndefined();
+    expect(res.json).toHaveBeenCalledWith({
+      status: "Email not in allowed email list",
+    });
+    expect(res.status).toHaveBeenCalledWith(401);
+  });
+
+  it("should allowed user with valid email to save", async () => {
+    process.env.ALLOWED_EMAIL_LIST = "@ons.gov.uk";
+    await upsertUser(req, res, next);
+    const user = await getUserByExternalId(req.user.externalId);
+    expect(user).toMatchObject(req.user);
+    expect(res.json).toHaveBeenCalledWith({ status: "OK" });
+  });
```

Comment:
Maybe throw in a test here for an empty email list, for example `process.env.ALLOWED_EMAIL_LIST = ""`. I guess if it is empty all emails should be allowed.

Reply:
Added test to generate user when the allowed_email_list is set to empty string